sycl/plugins/cuda/pi_cuda.hpp

```diff
@@ -678,6 +679,7 @@ struct _pi_kernel {
     pi_result retError = cuda_piKernelGetGroupInfo(
```

Comment:
Is calling cuda_piKernelGetGroupInfo needed here? Isn't it called by delegated constructor?

Reply:
There is not a strict requirement for cuda_piKernelGetGroupInfo to be called right here, some refactoring is possible. But the main idea of these changes was made for optimization purposes, so as not to call cuda_piKernelGetGroupInfo on each kernel submission.

Reply:
> LGTM

Does it mean you are okay to remove `cuda_piKernelGetGroupInfo` from this constructor?
It looks like it initializes `reqdThreadsPerBlock_`, which was already initialized in delegated constructor.

Reply:
Unless I'm missing something I don't believe this is being initialized in the delegated constructor. It probably could be moved though, or even maybe just used directly in `cuda_piEnqueueKernelLaunch`, as I believe it's only used there.

Reply:
Yes, it is okay.
PS: It seems that before I thought that  `_pi_kernel` is just a regular variable in the initialization list due to non-standard code style. Yes, since we have a deligated constructor, we can remove this piece of code.

Reply:
Since, @npmiller disagreed, I merged the fix w/o modification.
I'll leave it for Codeplay team to follow-up.
Just to clarify my point: I think [this code](https://github.com/intel/llvm/blob/sycl/sycl/plugins/cuda/pi_cuda.hpp#L678-L683) is redundant. According to my understanding it initializes `reqdThreadsPerBlock_` second time, the first time it was [here](https://github.com/intel/llvm/blob/sycl/sycl/plugins/cuda/pi_cuda.hpp#L667-L672).

Reply:
Ohh I see what you mean now, yeah it is duplicated, I was thinking about moving it to `cuda_piKernelCreate` instead so we can report any error properly.